### PR TITLE
[Modular] Cryostasis

### PR DESCRIPTION
**Pull request by unit0016**
## About The Pull Request
Adds Cryo sleepers to the game, which let people who need to leave.. leave, instead of just suiciding or SSDing. Also reopens job slots.
image
ShipStation Cryo:
image
image
LimaStation Cryo:
image

## Why It's Good For The Game
suicide bad mmmkay

## Changelog
:cl:
add: Cryosleep.
/:cl:

### CODE REVIEW

**Comment by Jolly-66** [comment]
Can you document where these will be in the maps?

File: jollystation_modules/code/game/machinery/cryopod.dm [hunk 1]
```diff
@@ -0,0 +1,487 @@
...
+	return control_computer != null
+
+/obj/machinery/cryopod/close_machine(mob/user)
+	if(!control_computer)
+		find_control_computer(TRUE)
+	if((isnull(user) || istype(user)) && state_open && !panel_open)
+		START_PROCESSING(SSmachines, src)
+		..(user)
+		var/mob/living/mob_occupant = occupant
+		if(mob_occupant && mob_occupant.stat != DEAD)
```
Comment (MrMelbert) [inline jollystation_modules/code/game/machinery/cryopod.dm:246]:
```suggestion
		if(mob_occupant?.stat != DEAD)
```

File: jollystation_modules/code/game/machinery/cryopod.dm [hunk 2]
```diff
@@ -0,0 +1,487 @@
...
+		return
+	if(occupant)
+		to_chat(user, "<span class='warning'>The cryo pod is already occupied!</span>")
+		return
+	if(target.stat == DEAD)
+		to_chat(user, "<span class='notice'>Dead people can not be put into cryo.</span>")
+		return
+	if(target.client && user != target)
+		if(iscyborg(target))
+			to_chat(user, "<span class='danger'>You can't put [target] into [src]. They're online.</span>")
```
Comment (MrMelbert) [inline jollystation_modules/code/game/machinery/cryopod.dm:435]: "They're online." lol

File: jollystation_modules/code/game/machinery/cryopod.dm [hunk 3]
```diff
@@ -0,0 +1,487 @@
...
+	else
+		visible_message("<span class='notice'>[user] starts putting [target] into the cryo pod.</span>")
+	if(do_after(user, 3 SECONDS, target = target))
+		if(occupant)
+			return
+		close_machine(target)
+		to_chat(target, "<span class='boldnotice'>If you ghost, log out or close your client now, your character will shortly be permanently removed from the round.</span>")
+		name = "[name] ([occupant.name])"
+		if(target == user)
+			log_admin("<span class='notice'>[key_name(target)] entered a stasis pod.</span>")
```
Comment (MrMelbert) [inline jollystation_modules/code/game/machinery/cryopod.dm:478]: `log_admin` seems like a weird place to put this, compared to `log_game` or something

**Review by MrMelbert — COMMENTED** [review]
i don't know how it's _supposed_ to work but it looks fine

**Review by MrMelbert — CHANGES_REQUESTED** [review]
also merge conflicts

File: jollystation_modules/code/game/machinery/cryopod.dm (hunk 2 above)
Comment (unit0016) [reply jollystation_modules/code/game/machinery/cryopod.dm:434]: It's talking about a borg. Seems fine to me.

**Review by MrMelbert — APPROVED** [review]
some improvements could still be made but nothing major so sure why not

File: jollystation_modules/code/game/machinery/cryopod.dm [hunk 4]
```diff
@@ -0,0 +1,486 @@
...
+	icon_state = "cryopod-open"
+	density = TRUE
+	anchored = TRUE
+	state_open = TRUE
+
+	var/on_store_message = "has entered long-term storage."
+	var/on_store_name = "Cryogenic Oversight"
+
+	// 15 minutes-ish safe period before being despawned.
+	var/time_till_despawn = 15 * 600 // This is reduced by 90% if a player manually enters cryo
```
Comment (MrMelbert) [inline jollystation_modules/code/game/machinery/cryopod.dm:176]: for future reference we have macros for time to use
`15 MINUTES` instead of `15 * 600`

**Pull request merged**